File: common_library/src/main/java/com/phone/common_library/adapter/TabFragmentStatePagerAdapter.java
package com.phone.common_library.adapter;

import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentStatePagerAdapter;

import org.jetbrains.annotations.NotNull;

import java.util.List;

/**
 * author    : Urasaki
 * e-mail    : 1164688204@qq.com
 * date      : 2020/3/27 11:16
 * introduce :
 */

public class TabFragmentStatePagerAdapter extends FragmentStatePagerAdapter {

    private static final String TAG = TabFragmentStatePagerAdapter.class.getSimpleName();
    private List<Fragment> fragmentList;

    public TabFragmentStatePagerAdapter(FragmentManager fm) {
        super(fm);
    }

    public TabFragmentStatePagerAdapter(FragmentManager fm, List<Fragment> fragmentList) {
        super(fm);
        this.fragmentList = fragmentList;
    }

    public void addFragment(Fragment fragment) {
    }

    @NotNull
    @Override
    public Fragment getItem(int position) {
        return fragmentList.get(position);
    }

    @Override
    public int getCount() {
        return fragmentList.size();
    }

    @NonNull
    @Override
    public Object instantiateItem(@NonNull ViewGroup container, int position) {
        return super.instantiateItem(container, position);
    }

    @Override
    public void destroyItem(@NonNull ViewGroup container, int position, @NonNull Object object) {
        super.destroyItem(container, position, object);
    }
}

File: library_common/src/main/java/com/phone/common_library/adapter/TabFragmentStatePagerAdapter.kt
package com.phone.common_library.adapter

import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentStatePagerAdapter

class TabFragmentStatePagerAdapter(fm: FragmentManager, var fragmentList: List<Fragment>) :
    FragmentStatePagerAdapter(fm) {

    private val TAG = TabFragmentStatePagerAdapter::class.java.simpleName

    fun addFragment(fragment: Fragment?) {}

    override fun getItem(position: Int): Fragment {
        return fragmentList[position]
    }

    override fun getCount(): Int {
        return fragmentList.size
    }

    override fun instantiateItem(container: ViewGroup, position: Int): Any {
        return super.instantiateItem(container, position)
    }

    override fun destroyItem(container: ViewGroup, position: Int, `object`: Any) {
        super.destroyItem(container, position, `object`)
    }
}